test/frontend.js

```diff
@@ -44,8 +44,52 @@ describe('Frontend', function () {
         .wait(() => document.getElementById('serverMsg').className.indexOf('danger') !== -1)
     })
   })
+  describe('Event Log Test', function () {
+    it('Should Append to the log', async function () {
+      this.timeout(12000)
+      await n.goto(page('event-log'))
+        .should.finally.containEql({ code: 200, url: page('event-log') })
+      let result = await n.wait(4000)
+        .evaluate((current, eventCount) => {
+          return { current: document.getElementById('LogPosition').innerText, eventCount: document.querySelectorAll('.event').length }
+        })
+      let obj = await n.click('#random')
+        .insert('textarea[name="payload"]', 'This is a test payment event')
+        .select('select[name="type"]', 'Payment')
+        .click('#submit')
+        .wait(4000)
```

Comment:
Instead of waiting for 4 seconds, I think you can simply pass in a function that waits for `document.querySelectorAll('.event').length >= 1` or whatever. This type of technique can then be used to get rid of all the other manual waits.